lib/eslint.js

```diff
@@ -648,6 +651,12 @@ module.exports = (function() {
             this.reset();
         }
 
+        if (typeof textOrParseResult === "string") {
+            text = textOrParseResult;
+        } else {
+            text = textOrParseResult.text || "";
```

Comment:
It doesn't. The object passed in is 
{ 
  text: original source
  ast: complete output from esprima
}
I'm open to suggestions on a better name for it.

I also considered using the esprima output with a text property added, but using a simple wrapper object is probably simpler.